packages/edit-navigation/src/components/menu-locations-editor/index.js

```diff
@@ -1,3 +1,122 @@
+/**
+ * External dependencies
+ */
+import { find, pick } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	Panel,
+	PanelBody,
+	SelectControl,
+	Spinner,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MenuSelectControl = ( { onChange, menus, menusList, location } ) => {
+	const [ menuId, setMenuId ] = useState( false );
+
+	return (
+		<>
+			{ menus && (
+				<SelectControl
+					options={ menusList }
+					value={ menuId || location.menu }
+					onChange={ ( selectedMenuId ) => {
+						setMenuId( selectedMenuId );
+						onChange( {
+							location: location.name,
+							menu: selectedMenuId,
+						} );
+					} }
+				/>
+			) }
+		</>
+	);
+};
+
 export default function MenuLocationsEditor() {
-	return <>Menu locations editor</>;
+	const menus = useSelect( ( select ) => select( 'core' ).getMenus() );
+	const menuLocations = useSelect( ( select ) =>
+		select( 'core' ).getMenuLocations()
+	);
+	const [ locationsData, setLocationsData ] = useState( {} );
+	const { saveMenu } = useDispatch( 'core' );
+
+	if ( ! menus || ! menuLocations ) {
+		return <Spinner />;
+	}
+
+	const setLocations = ( { location, menu } ) => {
+		locationsData[ location ] = menu;
+		setLocationsData( locationsData );
+	};
+
+	const saveLocations = async () => {
+		for ( const location of Object.keys( locationsData ) ) {
```

Comment:
This still doesn't feel right. An update of menu can take an array of menu location on update. Let take for example, a menu that is assigned to 3 locations. This update should be done in one, as not only is better performance, it is less likely to overwrite.

Reply:
Personally, I would loop through the locations genrating a object that looks like this 

```
{ 
   '123': ['social, 'footer'], 
   '456'  ['main']
}
```
Then navigate through this object performing the `saveMenu` calls.

Reply:
Or build an array of objects..

Reply:
but there is no UI to do that, is it?

Reply:
@draganescu Confused by the above question. The UI that has been created is the UI for this. You are just submitting it to the REST API in the wrong format.

Reply:
@spacedmonkey I modified the save function to send all locations for one menu in one call. Also fixed the function to respect await.